api/types/server_info.go

```diff
@@ -155,7 +155,10 @@ func (s *ServerInfoV1) MatchSearch(searchValues []string) bool {
 
 // GetNewLabels gets the labels to apply to matched Nodes.
 func (s *ServerInfoV1) GetNewLabels() map[string]string {
-	return s.Spec.NewLabels
+	if len(s.Spec.NewLabels) > 0 {
+		return s.Spec.NewLabels
+	}
+	return s.Metadata.Labels
```

Comment:
This is surprising behavior for a getter. I would prefer pushing this logic to the caller instead. If there are many callers we could consider adding a 3rd method, with a distinct name, that has this behavior.